loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,362 @@
+
+## File Format
+To allow for better clarity, the new mod file format will use the [JSON5 schema](https://json5.org/). This will allow for informative comments to be included in the default quilt.mod.json5 file as well as to allow mod/library developers to provide clarity for other developers.
+
+An example quilt.mod.json5:
+```json5
+{
+    schema_version: 1,
+    quilt_loader: {
+        group_id: "org.quiltmc",
+        mod_id: "example_mod",
+        version: "1.0.0",
+        entrypoints: {
+            main: [
+                "org.quiltmc.example_mod.impl.ExampleMod",
+                "org.quiltmc.example_mod.impl.ExampleModNetworking"
+            ],
+            // Since we only have a single client endpoint, no array is needed.
+            client: "org.quiltmc.example_mod.impl.client.ExampleModClient",
+        },
+        depends: {
+            "quilt_networking_api": "*",
+            "quilt_rendering_api": "*"
+        },
+        breaks: {
+            sodium: {
+                versions: "*",
+                reason: "Sodium does not implement the Quilt Rendering API."
+            },
+            some_random_library: [
+                "1.23.456", // A reason is not required
+                {
+                    versions: "<1.0.0",
+                    reason: "Stable API required"
+                },
+                {
+                    versions: "1.5.3",
+                    reason: "Contains a game-breaking bug"
+                }
+            ]
+        },
+        metadata: {
+            name: "Quilt Example Mod",
+            description: "An example mod for the Quilt ecosystem.",
+            contributors: {
+                "Haven King": "Developer"
+            },
+            contact: {
+                homepage: "https://quiltmc.org/"
+            },
+            license: "CC0-1.0",
+            icon: "assets/modid/icon.png"
+        }
+    },
+    mixins: [
+        "modid.mixins.json"
+    ]
+}
+```
+
+## Placeholders
+Strings with defined formats such as [the version field](#the-version-field) can instead supply a string matching the regex `^\$\{[a-zA-Z_$][a-zA-Z\d_$]*\}$` in a development environment to be replaced on build. The quilt-gradle plugin defines usage of the value `${version}` as a placeholder for the version declared in gradle.properties.
+
+# Drawbacks
+The primary drawback to this proposed format is the break from convention established by the Fabric project. It may make it more difficult for modders to adjust to the new toolchain if they are having to make drastic changes to their mod files.
+
+# Rationale and Alternatives
+It makes sense to take advantage of the breaking divergence of the toolchain to make changes that are better for Quilt's future. We could stick more strictly to the fabric.mod.json specification, but we would lose a lot of the advantages that this iteration has.
```

Comment:
I think that compatibility will be lost at some point anyway, and it is worth sacrificing for better features.